### ISSUE TITLE
App-module launched twice

### ISSUE THREAD

**Issue by froggiefab** [VeXell/pm2-prom-module#2]
Hi,

I'm trying to use your module (exactly as described in you article https://vexell.medium.com/pm2-module-to-monitoring-node-js-application-with-export-to-prometheus-and-grafana-43d4b958c563) but I must be doing something wrong somewhere.

The module's server is launched and visible right after `pm2 install pm2-prom-module` via `pm2 list`.

When I try to launch my app via `CMD ["pm2-runtime", "--json", "ecosystem.config.cjs", "--env", "$ENV_VERSION"]`, my app start just fine but pm2-prom-module try to launch another instance which result in the following outputs :
```
{"timestamp":"2024-03-25T12:47:58.119Z","type":"process_event","status":"exit","app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:47:58.126Z","type":"process_event","status":"online","app_name":"pm2-prom-module"}
{"message":"Error: listen EADDRINUSE: address already in use :::9988    at Server.setupListenHandle [as _listen2] (node:net:1872:16)    at listenInCluster (node:net:1920:12)    at Server.listen (node:net:2008:7)    at startPromServer (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:33:16)    at /root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:53:5    at PMX.initModule (/root/.pm2/modules/pm2-prom-module/node_modules/pmx/lib/index.js:97:12)    at Object.<anonymous> (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:35:15)    at Module._compile (node:internal/modules/cjs/loader:1376:14)    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)    at Module.load (node:internal/modules/cjs/loader:1207:32) {  code: 'EADDRINUSE',  errno: -98,  syscall: 'listen',  address: '::',  port: 9988}","timestamp":"2024-03-25T12:47:58.579Z","type":"err","process_id":1,"app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:47:58.595Z","type":"process_event","status":"exit","app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:47:58.602Z","type":"process_event","status":"online","app_name":"pm2-prom-module"}
{"message":"Error: listen EADDRINUSE: address already in use :::9988    at Server.setupListenHandle [as _listen2] (node:net:1872:16)    at listenInCluster (node:net:1920:12)    at Server.listen (node:net:2008:7)    at startPromServer (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:33:16)    at /root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:53:5    at PMX.initModule (/root/.pm2/modules/pm2-prom-module/node_modules/pmx/lib/index.js:97:12)    at Object.<anonymous> (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:35:15)    at Module._compile (node:internal/modules/cjs/loader:1376:14)    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)    at Module.load (node:internal/modules/cjs/loader:1207:32) {  code: 'EADDRINUSE',  errno: -98,  syscall: 'listen',  address: '::',  port: 9988}","timestamp":"2024-03-25T12:47:59.061Z","type":"err","process_id":1,"app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:47:59.077Z","type":"process_event","status":"exit","app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:47:59.084Z","type":"process_event","status":"online","app_name":"pm2-prom-module"}
{"message":"Error: listen EADDRINUSE: address already in use :::9988    at Server.setupListenHandle [as _listen2] (node:net:1872:16)    at listenInCluster (node:net:1920:12)    at Server.listen (node:net:2008:7)    at startPromServer (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:33:16)    at /root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:53:5    at PMX.initModule (/root/.pm2/modules/pm2-prom-module/node_modules/pmx/lib/index.js:97:12)    at Object.<anonymous> (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:35:15)    at Module._compile (node:internal/modules/cjs/loader:1376:14)    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)    at Module.load (node:internal/modules/cjs/loader:1207:32) {  code: 'EADDRINUSE',  errno: -98,  syscall: 'listen',  address: '::',  port: 9988}","timestamp":"2024-03-25T12:47:59.540Z","type":"err","process_id":1,"app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:47:59.556Z","type":"process_event","status":"exit","app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:47:59.563Z","type":"process_event","status":"online","app_name":"pm2-prom-module"}
{"message":"Error: listen EADDRINUSE: address already in use :::9988    at Server.setupListenHandle [as _listen2] (node:net:1872:16)    at listenInCluster (node:net:1920:12)    at Server.listen (node:net:2008:7)    at startPromServer (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:33:16)    at /root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:53:5    at PMX.initModule (/root/.pm2/modules/pm2-prom-module/node_modules/pmx/lib/index.js:97:12)    at Object.<anonymous> (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:35:15)    at Module._compile (node:internal/modules/cjs/loader:1376:14)    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)    at Module.load (node:internal/modules/cjs/loader:1207:32) {  code: 'EADDRINUSE',  errno: -98,  syscall: 'listen',  address: '::',  port: 9988}","timestamp":"2024-03-25T12:48:00.027Z","type":"err","process_id":1,"app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:48:00.044Z","type":"process_event","status":"exit","app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:48:00.051Z","type":"process_event","status":"online","app_name":"pm2-prom-module"}
{"message":"Error: listen EADDRINUSE: address already in use :::9988    at Server.setupListenHandle [as _listen2] (node:net:1872:16)    at listenInCluster (node:net:1920:12)    at Server.listen (node:net:2008:7)    at startPromServer (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:33:16)    at /root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:53:5    at PMX.initModule (/root/.pm2/modules/pm2-prom-module/node_modules/pmx/lib/index.js:97:12)    at Object.<anonymous> (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:35:15)    at Module._compile (node:internal/modules/cjs/loader:1376:14)    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)    at Module.load (node:internal/modules/cjs/loader:1207:32) {  code: 'EADDRINUSE',  errno: -98,  syscall: 'listen',  address: '::',  port: 9988}","timestamp":"2024-03-25T12:48:00.509Z","type":"err","process_id":1,"app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:48:00.525Z","type":"process_event","status":"exit","app_name":"pm2-prom-module"}
{"timestamp":"2024-03-25T12:48:00.531Z","type":"process_event","status":"online","app_name":"pm2-prom-module"}
{"message":"Error: listen EADDRINUSE: address already in use :::9988    at Server.setupListenHandle [as _listen2] (node:net:1872:16)    at listenInCluster (node:net:1920:12)    at Server.listen (node:net:2008:7)    at startPromServer (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:33:16)    at /root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:53:5    at PMX.initModule (/root/.pm2/modules/pm2-prom-module/node_modules/pmx/lib/index.js:97:12)    at Object.<anonymous> (/root/.pm2/modules/pm2-prom-module/node_modules/pm2-prom-module/index.js:35:15)    at Module._compile (node:internal/modules/cjs/loader:1376:14)    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)    at Module.load (node:internal/modules/cjs/loader:1207:32) {  code: 'EADDRINUSE',  errno: -98,  syscall: 'listen',  address: '::',  port: 9988}","timestamp":"2024-03-25T12:48:01.011Z","type":"err","process_id":1,"app_name":"pm2-prom-module"}
```
Do you have any idea why pm2 try to launch another instance ? Is there a setting I'm missing to prevent that behavior ?

Thanks

**Comment by VeXell** [comment]
Hello @froggiefab 

Thank you for you message.

What PM2 version do you use ?

I run `pm2-prom-module` also in docker and did not face problem like that.

My config:

```
pm2 install pm2-prom-module && pm2 set pm2-prom-module:port $ENV_METRICS_PORT && pm2 set pm2-prom-module:service_name $PROJECT_NAME

CMD ["pm2-runtime", "--json", ".ecosystem.config.js"]
```

Could you try to use another port?

**Comment by froggiefab** [comment]
Hi,

Thanks for the quick response.

I'm was using the latest version but tried 5.2.2 with no luck.
I also tried with another port and ended up with the same result. 

The main difference I see between our configurations is that I use a "boilerplate" image to initialize pm2 + pm2-prom-module with an entrypoint that execute `pm2 set pm2-prom-module:service_name $APP_NAME-metrics`. 
But even when I tried to remove this command, I get the same error.

I finally tried with a "FROM" instruction pointing at a "vanilla" node install and this time everything worked fine. So it's finally a Docker problem on my part.

Sorry for bothering you.

Thank you for your time and for this great project :)
